modules/properties-config/src/main/java/org/mule/runtime/config/api/properties/ConfigurationPropertiesHierarchyBuilder.java

```diff
@@ -188,4 +187,26 @@ public ConfigurationPropertiesResolver build() {
     lastResolver.setAsRootResolver();
     return lastResolver;
   }
+
+  @Deprecated
```

Comment:
We usually put the annotations after the javadoc.
Also, add the javadoc for this deprecation